src/plugins/intel_gpu/src/graph/include/moe_inst.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "intel_gpu/primitives/moe_fused_compressed.hpp"
+#include "primitive_inst.h"
+
+namespace cldnn {
+namespace details {}
+
+template <>
+struct typed_program_node<moe_fused_compressed> : public typed_program_node_base<moe_fused_compressed> {
```

Comment:
File name is  too general: moe_inst.h 
Pleae rename all the relevant primitives, test names, inst, node name to 
moe_fused_3gemm_swiglu